design/stackhub.md

```diff
@@ -0,0 +1,207 @@
+  -  Object model inheritence, e.g. A Collection ISA a Stack and a CollectionHub ISA a Stackhub, causes confusion especially for community messsaging, developers and new comers to the project.
+  -  Monolithic build (CollectionHub) causes painful lifecycle for making configuration or manage the curated stacks and pipelines.
+
+- In general, to replace Collection Hubs with Stack Hubs, and providing configuration for the governed stacks via the defined Kubernetes Custom Resources:  Kabanero CR.
+
+- Renaming the collection CR a stack CR, which has essentially the same data and format.
+
+- Although this is not a seemingly drastic change -- the concept of CollectionHub and Collections has tentacles throughout the website, guides, documentation, videos and blogs.
+
+## User stories
+
+- As the Kabanero community, we would like to simplify the messaging of Kabanero by removing the concept of CollectionHub and Collections.
+
+- As Champ (architect), I would like the definition, selection and build of Appsody stacks that are "blessed" untangled from the configuration of a specific Kabanero instance, moving from the Collection-Hub concept to a strictly Kubernetes-natural Kabanero and Stack CRD gitops user experience.
+
+## As-is
+
+- To introduce Kabanero topics, one must teach the student the concept of stacks, stack-hubs, pipelines, application development.  After all of that, then one must teach the student about collections and collection hubs, with a collection being a stack and a collection hub being a stack hub.  
+
+- To customize a stack, champ uses the appsody tools to copy the stack, and then has to manually convert it into a collection and include it in a collection hub.
+
+- To customize one character of a pipeline, I need to do a complete collection hub rebuild, which depending on how developer's are consuming collection hub releases may cause them to need to update all of their configuration -- for a change doesn't directly.
+
+## To-be
+-  Champ's (architect's) lifecycle changes to build stacks, using Appsody stackhub builds and independently building pipeline releases.
+-  Champ's experience is Kubernetes and operator-centric.  The configuration for Kabanero is defined solely by Kubernetes resources: `Kind:Kabanero` and `Kind:Stack`.
+-  Kabanero concepts are simplified to using stacks and their configuration.
+
+## Main Feature design
+
+### Custom Resource Changes
+
+#### Kind:Kabanero
+
+The Kabanero CRD will change to specifiy a Stack hub URL instead of a Collection Hub URL.  Previously, the Kabanero CRD did not concern itself with the relationship of pipelines with stacks.  The Kabanero CRD is enhanced to provide a default pipelines URL. 
+
+#### Kind:Stack
+
+The existing schema of the Collections CRD will be copied, with the `Kind:` field changing to `Stack`.  All other fields are relevant with one expection.  The status field `collections.url` field is being removed.  (Rationale below.)
```

Comment:
The `Collection` CRD did not contain fields for pipelines or images.  We've added that to the `Stack` CRD, since the information is no longer present in the collection hub index.  Do we need to call that out here?

Reply:
We need to understand if this is the right decision, for an unmanaged stack

Reply:
One change we decided in a discussion with @kaczyns is to pair the index.yaml URL specification in the Kabanero UR with a pipeline.yaml; allowing overrides to occur within the Kabanero CR.  I'll be changing the design proposal today with those changes.

For 0.6.0, we are trying to not depend on any Appsody changes as a form of incremental improvements.  We expect Champ's (architect) ability to have the flexibility to build unique index.yamls will be improved to the point that using an unmanaged stack will be unusual.  

That said, we cannot keep Champ from `oc apply stack.yaml`, so we at least have to define what happens when that happens-- having the index.yaml file in the Stack.yaml actually got confusing when the index.yaml file itself could have 1:n stacks in it.  Since Champ was providing all of the detail in the stack.yaml CR in the unmanaged case, it caused an issue where the data in the CR didn't match the url.